### PR TITLE
Add S3 public access block config fetching

### PR DESCRIPTION
**Pull request by ari-aviran**
### Summary of your changes
Add logic to fetch S3 public access block configuration for account and buckets.

### Screenshot/Data
Screenshot 2023-04-03 at 10 06 18

### Related Issues
Related https://github.com/elastic/csp-security-policies/issues/194

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Comment by ari-aviran** [comment]
Need to update policy version after https://github.com/elastic/csp-security-policies/issues/194 is merged

File: resources/providers/awslib/s3/provider.go [hunk 1]
```diff
@@ -58,11 +64,21 @@ func (p Provider) DescribeBuckets(ctx context.Context) ([]awslib.AwsResource, er
...
 	var result []awslib.AwsResource
+
+	if len(clientBuckets.Buckets) == 0 {
+		return result, nil
+	}
+
+	accountPublicAccessBlockConfig, getAccountPublicAccessBlockErr := p.getAccountPublicAccessBlock(ctx)
+	if getAccountPublicAccessBlockErr != nil {
+		p.log.Errorf("Could not get account public access block configuration. Err: %v", getAccountPublicAccessBlockErr)
+	}
```
Comment (uri-weisman) [inline resources/providers/awslib/s3/provider.go:75]:
```suggestion
	accountPublicAccessBlockConfig, accountPublicAccessBlockErr := p.getAccountPublicAccessBlock(ctx)
	if accountPublicAccessBlockErr != nil {
		p.log.Errorf("Could not get account public access block configuration. Err: %v", accountPublicAccessBlockErr)
	}
```

**Review by uri-weisman — APPROVED** [review]
Approved with some minor comments

File: resources/providers/awslib/s3/s3.go [hunk 2]
```diff
@@ -20,15 +20,18 @@ package s3
...
-	Name             string            `json:"name"`
-	SSEAlgorithm     *string           `json:"sse_algorithm,omitempty"`
-	BucketPolicy     BucketPolicy      `json:"bucket_policy,omitempty"`
-	BucketVersioning *BucketVersioning `json:"bucket_versioning,omitempty"`
+	Name                                  string                          `json:"name"`
+	SSEAlgorithm                          *string                         `json:"sse_algorithm,omitempty"`
+	BucketPolicy                          BucketPolicy                    `json:"bucket_policy,omitempty"`
+	BucketVersioning                      *BucketVersioning               `json:"bucket_versioning,omitempty"`
+	PublicAccessBlockConfiguration        *PublicAccessBlockConfiguration `json:"public_access_block_configuration"`
+	AccountPublicAccessBlockConfiguration *PublicAccessBlockConfiguration `json:"account_public_access_block_configuration"`
```
[the remark this answers is not in the document]
Comment (ari-aviran) [reply resources/providers/awslib/s3/s3.go]: No reason I guess

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**